File: backend/semgrep_rules/tests/javascript/kisa_sw_42_uninitialized_variable.js
function unsafe(use) {
  let value;
  // ruleid: kisa.sw42.javascript.uninitialized-variable-use-review
  use(value);
}
function safe(use) {
  let value = "initialized";
  // ok: kisa.sw42.javascript.uninitialized-variable-use-review
  use(value);
}

File: backend/semgrep_rules/tests/java/kisa_sw_42_uninitialized_variable.java
class kisa_sw_42_uninitialized_variable {
    void unsafe(Consumer c) {
        String value;
        value = null;
        // ruleid: kisa.sw42.java.declaration-without-initializer-review
        c.accept(value);
    }
    void safe(Consumer c) {
        String value = "initialized";
        // ok: kisa.sw42.java.declaration-without-initializer-review
        c.accept(value);
    }
    interface Consumer { void accept(String v); }
}
